### PR TITLE
[perf] prepare_prefill_qkv hook + fp8 quantize jit kernel

### PR DESCRIPTION
**Pull request by Qiaolin-Yu**
Adds a model-side `prepare_prefill_qkv` extension point on the MLA attention backends so backends can quantize/pack/rope Q/K/V *before* the prefill kernel call, and ships a tokenspeed_mla-side fp8 quantize jit kernel as one such producer:

- python/sglang/jit_kernel/fp8_quantize.py (new) — fp8 e4m3 quantize helper used by the new prepare path.
- python/sglang/srt/layers/attention/tokenspeed_mla_backend.py — prepare_prefill_qkv path that lands FP8-quantized Q/K/V into the tokenspeed_mla prefill kernel (BF16 Q/K/V → FP8 single shot, no model-side staging).
- python/sglang/srt/models/deepseek_common/attention_forward_methods/ forward_mha.py — call `backend.prepare_prefill_qkv(...)` when the active attention backend exposes it; otherwise fall through to the existing path.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #26148765775](https://github.com/sgl-project/sglang/actions/runs/26148765775)
Latest PR Test (Extra): :white_check_mark: [Run #26149408054](https://github.com/sgl-project/sglang/actions/runs/26149408054)

### CODE REVIEW

File: python/sglang/srt/layers/attention/trtllm_mla_backend.py [hunk 1]
```diff
@@ -1144,7 +1144,7 @@ def forward_extend(
             assert k_rope is None
             chunk_idx = forward_batch.prefix_chunk_idx
 
-            out = torch.zeros(
+            out = torch.empty(
```
Comment (kpham-sgl) [inline python/sglang/srt/layers/attention/trtllm_mla_backend.py:1147]: curious what is this change for?

File: python/sglang/srt/models/deepseek_common/attention_forward_methods/forward_mha.py [hunk 2]
```diff
@@ -220,6 +220,24 @@ def forward_normal_prepare(
             kv_a = self.kv_a_layernorm(kv_a)
 
         k_pe = latent_cache[:, :, self.kv_lora_rank :]
+
+        # Backend prefill hook: the backend owns the BF16->FP8 transition
+        # (fused RoPE + quantize for Q/K, direct FP8 KV-cache write) and
+        # returns FP8 tensors ready for its kernel. Backends without the
+        # hook fall through to the BF16 path below.
+        backend = _resolve_attn_backend(forward_batch)
+        if hasattr(backend, "prepare_prefill_qkv"):
```
Comment (kpham-sgl) [inline python/sglang/srt/models/deepseek_common/attention_forward_methods/forward_mha.py:229]: Right now only `tokenspeed_mla` have `prepare_prefill_qkv` right? Is this something `trtllm_mla` _should_ have also?

**Review by kpham-sgl — APPROVED** [review]

File: python/sglang/srt/layers/attention/trtllm_mla_backend.py (hunk 1 above)
Comment (kpham-sgl) [reply python/sglang/srt/layers/attention/trtllm_mla_backend.py:1147]: nvm this avoid a fill functor right? and we are confident its correct?

File: python/sglang/srt/models/deepseek_common/attention_forward_methods/forward_mha.py (hunk 2 above)
Comment (Qiaolin-Yu) [reply python/sglang/srt/models/deepseek_common/attention_forward_methods/forward_mha.py:229]: yes. sure, make trtllm_mla use the same logic should be in the future work. actually, I think since flashinfer has integrated this kernel in trtllm backend, we will merge tokenspeed_mla backend into existing trtllm_mla backend

File: python/sglang/srt/layers/attention/trtllm_mla_backend.py (hunk 1 above)
Comment (Qiaolin-Yu) [reply python/sglang/srt/layers/attention/trtllm_mla_backend.py:1147]: This is from @ch-wan , to eliminate two `zeros` operations.

File: python/sglang/srt/models/deepseek_common/attention_forward_methods/forward_mha.py (hunk 2 above)
Comment (kpham-sgl) [reply python/sglang/srt/models/deepseek_common/attention_forward_methods/forward_mha.py:229]: nit: can we leave a todo so we don't forget about this in the future? Especially in near future when `trtllm_mla` will match `tokenspeed_mla` perf

Comment (Qiaolin-Yu) [reply python/sglang/srt/models/deepseek_common/attention_forward_methods/forward_mha.py:229]: sure

Comment (Qiaolin-Yu) [reply python/sglang/srt/models/deepseek_common/attention_forward_methods/forward_mha.py:229]: done
